Python/context.c

```diff
@@ -880,7 +880,7 @@ contextvar_new(PyObject *name, PyObject *def)
 
     var->var_hash = contextvar_generate_hash(var, name);
     if (var->var_hash == -1) {
-        Py_DECREF(var);
```

Comment:
It introduces a leak:

```
$ ./python -m test test_context -v -m test_context_new_unhashable_str_subclass -R 3:3
...
test_context leaked [1, 1, 1] references, sum=3
...
```

Instead, can you move the code which initialize *var_hash* at the end, and keep Py_DECREF()?